framework/grid/ActionColumn.php

```diff
@@ -85,6 +85,28 @@ class ActionColumn extends Column
      * ```
      */
     public $buttons = [];
+    /** @var array visibility conditions for each button. The array keys are the button names (without curly brackets),
+     * and the values are the conditions whether this button is visible.Can be boolean value or callback. Defaults to true.
+     * The callbacks should use the following signature:
+     *
+     * ```php
+     * [
+     *     'update' => function ($model) {
+     *         return $model->status === 'editable';
+     *     },
+     * ],
+     * ```
+     *
+     * Or you can pass a boolean expression:
+     *
+     * ```php
+     * [
+     *     'update' => \Yii::$app->user->can('update'),
+     * ],
+     * ```
+     */
+    public $buttonsVisible = [];
```

Comment:
I don't really like this name because it breaks rules about words order in English. Let's rename this property to `visibleButtons` instead